internal/provider/service_account_resource.go

```diff
@@ -0,0 +1,252 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	unleash "github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource                = &serviceAccountResource{}
+	_ resource.ResourceWithConfigure   = &serviceAccountResource{}
+	_ resource.ResourceWithImportState = &serviceAccountResource{}
+)
+
+func NewServiceAccountResource() resource.Resource {
+	return &serviceAccountResource{}
+}
+
+type serviceAccountResource struct {
+	client *unleash.APIClient
+}
+
+type serviceAccountResourceModel struct {
+	Id       types.Int64  `tfsdk:"id"`
+	Name     types.String `tfsdk:"name"`
+	UserName types.String `tfsdk:"username"`
+	RootRole types.Int64  `tfsdk:"root_role"`
+}
+
+func (r *serviceAccountResource) Configure(ctx context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*unleash.APIClient)
+	if !ok {
+		tflog.Error(ctx, "Unable to prepare client")
+		return
+	}
+	r.client = client
+}
+
+func (r *serviceAccountResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account"
+}
+
+func (r *serviceAccountResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages a service account.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Description: "The ID of the service account.",
+				Computed:    true,
+			},
+			"username": schema.StringAttribute{
+				Description: "The username for the service account.",
+				Optional:    true,
```

Comment:
Maybe we can make it required for terraform:
```suggestion
				Required:    true,
```

Reply:
Cheers, suggestion taken.

Turns out it's required on the API level too. So you get blocked by the validation if it's not set

Name on the other hand is marked as required on the UI. Not required on the API and not required on the OpenAPI spec

:upside_down_face: